Add troubleshooting points for join issues (#31)
CyberEyeResearchLabs/lorawan-stack-docs: doc/content/devices/troubleshooting/_index.md

### My device won't join. What do I do?

- Double check your DevEUI, JoinEUI or AppEUI, LoRaWAN and Regional Parameters Version, root keys (AppKey, and with LoRaWAN 1.1 or higher, NwkKey)

> - Check gateway and application events for traffic from your device. Please find some of the common errors below.

Reviewer: ```suggestion
- Check gateway and device events for traffic from your device. Below are a few common issues.
```

>     - **MIC mismatch** in the device events: The possible cause is the mismatch of AppKey in device firmware to the AppKey registered in The Things Stack console.

Reviewer: ```suggestion
    - **MIC mismatch** error in the device events: The possible cause is the mismatch of AppKey in device firmware to the AppKey registered in The Things Stack console. Update the AppKey in the console accordingly.
```

>     - **DevNonce has already been used** in the device events: It generally happens when the device had sent many unsuccessful Join Requests. To reset the DevNonces, please store the current AppKey of the device and try changing the device AppKey in the console. The backend clears the DevNonces, and then you can revert to the previous AppKey. The device should now join successfully without this error.

Reviewer: ```suggestion
    - **DevNonce has already been used** warning in the device events: It generally happens when the device had sent too many unsuccessful Join Requests.
```

>     - **uplink channel not found** in the device events: It indicates the device is transmitting Join Requests in the non-default channels of the band which is not in line with the LoRaWAN Specification. We would suggest contacting the end-device manufacturer in this regard.

Reviewer: ```suggestion
    - **uplink channel not found** error in the device events: It indicates the device is transmitting Join Requests in the non-default channels of the band and it is not as per the LoRaWAN Specification. Check device firmware in this regard.
```

>     - **uplink channel not found** in the gateway events: It indicates there is a mismatch of the frequency plan. Configure the same frequency plan in Device settings and Gateway settings of The Things Enterprise Stack console.

Reviewer: ```suggestion
    - **uplink channel not found** error in the gateway events: It indicates there is a mismatch of the frequency plans. Configure the same frequency plan in Device settings and Gateway settings of The Things Stack console.
```

- Double check frequency plan settings in your end device and gateways (they must be the same LoRaWAN band)
- Double check your network connection. If there is a slow connection from the server to the gateway, the join accept message may be sent too late (this can happen when a gateway uses 3G as a backhaul). If using the CLI, run `ttn-lw-cli gateways connection-stats <gateway-id>` to see the round trip time (RTT) for your gateway

> - Check for duplicate use of JoinNonce (or AppNonce) and/or DevNonce

Reviewer: This is redundant with the point we added above
